src/duckdb/src/main/relation.cpp

```diff
@@ -324,7 +324,9 @@ unique_ptr<QueryResult> Relation::Query(const string &sql) {
 }
 
 unique_ptr<QueryResult> Relation::Query(const string &name, const string &sql) {
-	CreateView(name);
+    bool replace = true;
+    bool temp = IsReadOnly();
+	CreateView(name, replace, temp);
```

Comment:
@Tmonster: I realized too late that this patches the duckdb sources. Has this been upstreamed by now? Do you want to?

I can work around with a patch file for now.

Reply:
has not been upstreamed yet, and might not be until after v1.0 comes out. I think a patch file in duckdb-r is the way to go for now. I will still create an upstream PR though so you can eventually remove that patch.